src/PKSim.Infrastructure/Services/ObservedDataTask.cs

```diff
@@ -62,19 +62,23 @@ public void ExportToPkml(DataRepository observedData)
          _observedDataPersistor.Save(observedData, file);
       }
 
-      public void AddObservedDataToAnalysable(DataRepository observedData, IAnalysable analysable)
+      public void AddObservedDataToAnalysable(IReadOnlyList<DataRepository> observedData, IAnalysable analysable)
       {
          AddObservedDataToAnalysable(observedData, analysable, showData: false);
       }
 
-      public void AddObservedDataToAnalysable(DataRepository observedData, IAnalysable analysable, bool showData)
+      public void AddObservedDataToAnalysable(IReadOnlyList<DataRepository> observedData, IAnalysable analysable, bool showData)
       {
          var simulation = analysable as Simulation;
-         if (simulation == null) return;
+         if (simulation == null)
+            return;
+
+         var observedDataToAdd = observedData.Where(x => !simulation.UsesObservedData(x)).ToList();
+         if (!observedDataToAdd.Any())
+            return;
 
-         if (simulation.UsesObservedData(observedData)) return;
-         simulation.AddUsedObservedData(observedData);
-         _executionContext.PublishEvent(new ObservedDataAddedToAnalysableEvent(simulation, observedData, showData));
+         observedDataToAdd.Each(simulation.AddUsedObservedData);
+         _executionContext.PublishEvent(new ObservedDataAddedToAnalysableEvent(simulation, observedDataToAdd, showData));
```

Comment:
I have changed the interfaces in Core to use collection of observeddata instead of only one. This allow for a massive boost in performance that ws not possible before with the old chart implementation